fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -631,7 +631,7 @@ public CommandProcessingResult closeClient(final Long clientId, final JsonComman
                 throw new InvalidClientStateTransitionException("close", "is.under.transfer", errorMessage);
             }
 
-            if (client.isNotPending() && client.getActivationLocalDate().isAfter(closureDate)) {
+            if (client.isNotPending() && client.getActivationLocalDate() != null && client.getActivationLocalDate().isAfter(closureDate)) {
```

Comment:
Instead of checking activation date to null why can't we add a check if client status is active and client activation date is after closure date?